tracing/traced_file.cc

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <seastar/core/future.hh>
+#include <seastar/core/print.hh>
+
+#include "tracing/traced_file.hh"
+
+using namespace seastar;
+
+class traced_file_impl : public file_impl {
+private:
+    file _f;
+
+    tracing::trace_state_ptr _trace_state;
+    sstring _trace_prefix;
+
+public:
+    unsigned _memory_dma_alignment;
+    unsigned _disk_read_dma_alignment;
+    unsigned _disk_write_dma_alignment;
+
```

Comment:
file_impl already contains these variables, and file::disk_read_dma_alignment() will talk to them. So these should be dropped on the file_impl members initialized instead.

Reply:
Brain fart, sorry. Fixed.